addons/mass_mailing/models/mailing.py

```diff
@@ -488,8 +498,19 @@ def action_test(self):
         }
 
     def action_launch(self):
-        self.write({'schedule_type': 'now'})
-        return self.action_put_in_queue()
+        mailing_send_immediate_nbr = self.env['ir.config_parameter'].sudo().get_param('mass_mailing.mass_mailing_immediate_send_nbr', 20)
+        if len(self._get_recipients()) > int(mailing_send_immediate_nbr):
```

Comment:
action_launch is technically a multi so you should probably split into two record sets, one to put in queue and one for send_mail.